### PR TITLE
Instructions and scripts for daily and monthly Helix Core backups

### PR DESCRIPTION
**Pull request by davidair**

### CODE REVIEW

File: backups/README.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+
+To create an offline root directory:
+
+1. Create a new directory on a disk which has enough storage to hold metadata. Note that the script
+expects the server id to be part of that path. For example, if the server is named "demo", an
+offline root directory could be /opt/perforce/offline:
+
+```
+mkdir -p /opt/perforce/offline/demo
+```
```
Comment (noahdungey) [inline backups/README.md:134]: Do you want to add a chmod line here to remove world-readable access incase of sensitive files.

File: backups/README.md [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+can name the Incremental bucket *incremental-my-perforce-123*.
+
+There are a number of considerations for choosing the location type and storage class.
+We recommend reviewing Google Cloud documentation to understand the tradeoffs involved in
+making those choices. For example, a Multi-region location type would incur the highest cost
+but will provide the highest availability. A Nearline storage class could be appropriate for
+backups, but you would need to evaluate this based on cost and performance characteristics.
+
+We recommend configuring a uniform access to all objects in the bucket.
+
```
Comment (noahdungey) [inline backups/README.md:73]: Recommend adding a note about verifying that bucket is not publicly readable, due to sensitive data.

File: backups/monthly_backup.sh [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  local depot_path=$2
+  local dry_run=$3
+  local is_edge=$4
+
+  # Backing up versioned files
+  log "Backing up the versioned files at ${depot_path}"
+
+  if [[ "${is_edge}" -eq 0 ]]; then
+    gsutil \
+      -m cp -r "${depot_path}" 2>&1 \
```
Comment (noahdungey) [inline backups/monthly_backup.sh]: Why do you have multiple STDERR into STDOUT redirections inside a single command?

**Review by noahdungey — COMMENTED** [review]
Looks good overall, but I'd like to see more information about data sensitivity to some of the dumped Perforce tables.

File: backups/README.md (hunk 1 above)
Comment (davidair) [reply backups/README.md:134]: Added chmod 700 /opt/perforce/offline

File: backups/README.md (hunk 2 above)
Comment (davidair) [reply backups/README.md:73]: Added a note.

File: backups/monthly_backup.sh (hunk 3 above)
Comment (davidair) [reply backups/monthly_backup.sh]: Why, doesn't it redirect better if it has multiple ones? :) Fixed!

File: backups/monthly_backup.sh [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+#######################################
+# Parses options and optionally prints usage.
+# Globals:
+#   USAGE
+# Arguments:
+#   Commnand-line arguments
+#######################################
+parse_options() {
+  set +e
+  OPTS="$(getopt -o "d,e" --long "dry_run,logs_root:,is_edge,help" -n "$(basename "$0")" -- "$@")"
```
Comment (noahdungey) [inline backups/monthly_backup.sh]: clean up quoting.

File: backups/nightly_backup.sh [hunk 5]
```diff
@@ -0,0 +1,454 @@
...
+#######################################
+# Parses options and optionally prints usage.
+# Globals:
+#   USAGE
+# Arguments:
+#   Command-line arguments
+#######################################
+parse_options() {
+  set +e
+  OPTS="$(getopt -o "d,e" --long "dry_run,logs_root:,is_edge,help" -n "$(basename "$0")" -- "$@")"
```
Comment (noahdungey) [inline backups/nightly_backup.sh]: Clean up quoting

File: backups/nightly_backup.sh [hunk 6]
```diff
@@ -0,0 +1,454 @@
...
+get_latest_checkpoint () {
+  local server_id=$1
+  local checkpoints_path=$2
+  echo "$(find "${checkpoints_path}" -name "${server_id}.ckp.*" -type f -printf '%T@ %p\n' | sort -n | grep -v ".md5$" | tail -1 | cut -f2- -d" ")"
+}
+
+get_latest_journal () {
+  local server_id=$1
+  local checkpoints_path=$2
+  echo "$(find "${checkpoints_path}" -name "${server_id}.jnl.*" -type f -printf '%T@ %p\n' | sort -n | grep -v ".md5$" | tail -1 | cut -f2- -d" ")"
```
Comment (noahdungey) [inline backups/nightly_backup.sh]: check quoting

**Review by noahdungey — COMMENTED** [review]
Found a couple more places where quoting can be cleaned up.

File: backups/monthly_backup.sh (hunk 4 above)
Comment (davidair) [reply backups/monthly_backup.sh]: Done

File: backups/nightly_backup.sh (hunk 6 above)
Comment (davidair) [reply backups/nightly_backup.sh]: Done

File: backups/nightly_backup.sh (hunk 5 above)
Comment (davidair) [reply backups/nightly_backup.sh]: Done

**Pull request merged**